File: 347-TopKFrequentElements/347-TopKFrequentElements.py
# Last updated: 2/11/2026, 11:08:24 AM
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        mp = {}

        for num in nums:
            mp[num] = mp.get(num,0) + 1
            
        buckets = [[] for _ in range(len(nums)+1)]
        
        for key,value in mp.items():
            buckets[value].append(key)
            
        res = []
        current_count = 0
            
        for i in range(len(nums),0,-1):
            for key in buckets[i]:
                if (current_count < k):
                    current_count+=1
                    res.append(key)
        
        return res



        

File: 347-TopKFrequentElements/347-TopKFrequentElements.cpp
// Last updated: 2/11/2026, 11:08:22 AM
#include <unordered_map>
#include <vector>

using namespace std;


class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        unordered_map<int,int> mp;
        
        for (auto i: nums){
            mp[i]++;
        }

        vector<vector<int>> arr (nums.size()+1);

        for (auto &[key, value]: mp){
            arr[value].push_back(key);
        }

        vector<int>res;
        int count = 0;

        for (int i = nums.size(); count < k && i >= 0; --i){
            for (auto v: arr[i]){
                count++;
                res.push_back(v);
            }
        }

        return res;

    }
};